lib/src/connection.dart

```diff
@@ -378,17 +377,20 @@ abstract class _PostgreSQLExecutionContextMixin implements PostgreSQLExecutionCo
     }).toList();
   }
 
-  Future _resolveTableOIDs(List<int> oids) async {
+  Future _resolveTableOIDs(Set<int> oids) async {
     final unresolvedIDString = oids.join(",");
-    final orderedTableNames =
-        await query("SELECT relname FROM pg_class WHERE relkind='r' AND oid IN ($unresolvedIDString) ORDER BY oid ASC");
-
-    final iterator = oids.iterator;
-    orderedTableNames.forEach((tableName) {
-      iterator.moveNext();
-      if (tableName.first != null) {
-        _tableOIDNameMap[iterator.current] = tableName.first;
-      }
+    final orderedTableNames = await query(
+        "SELECT oid, relname FROM pg_class WHERE relkind='r' AND oid IN ($unresolvedIDString) ORDER BY oid ASC");
+
+    for (List list in orderedTableNames) {
+      final int oid = list[0];
```

Comment:
Tests are failing on this line because `list[0]` appears to have different types (tests show String and Uint8array. The general purpose approach you've made here makes sense. I've verified that an OID of 0 basically means something that isn't permanent, e.g. result of a function.

Reply:
Not sure what's the best way to fix this. Maybe RIP this PR and create a new one focusing only on OID=0?

Reply:
I think so - we shouldn't even go to the resolution stage for OIDs of 0. Does it make sense to filter 0s out of the original set before even calling this method?